src/Compilers/CSharp/Portable/BoundTree/BoundExpression.cs

```diff
@@ -3,15 +3,34 @@
 using System.Collections.Immutable;
 using System.Diagnostics;
 using Microsoft.CodeAnalysis.CSharp.Symbols;
-using Microsoft.CodeAnalysis.CSharp.Syntax;
-using Microsoft.CodeAnalysis.Text;
 using Roslyn.Utilities;
 using System;
 
 namespace Microsoft.CodeAnalysis.CSharp
 {
     internal partial class BoundExpression
     {
+        protected abstract BoundExpression DefaultShallowClone();
```

Comment:
>DefaultShallowClone [](start = 43, length = 19)

Do we need both virtual methods `DefaultShallowClone()` and `ShallowClone()`? Could we collapse to one method if the method is only generated conditionally, say unless `ExternalShallowClone="true"` is set on the type in BoundNodes.xml?